tests/tt_metal/tt_metal/unit_tests/multichip/erisc_app_direct_send.cpp

```diff
@@ -17,13 +17,10 @@
 #include "tt_metal/test_utils/print_helpers.hpp"
 #include "tt_metal/test_utils/stimulus.hpp"
 
+namespace ANON_NAMESPACE {
```

Comment:
It is weird to give a namespace such a name. I’d like to understand your thinking process here

Reply:
Renamed to CMAKE_UNIQUE_NAMESPACE

Reply:
If the UNITY_BUILD is enabled, CMake will replace `CMAKE_UNIQUE_NAMESPACE` with a unique identifier. `namespace {` is still needed to make the namespaces non-overlapping in case UNITY_BUILD is turned off (see [their example](https://cmake.org/cmake/help/latest/prop_tgt/UNITY_BUILD_UNIQUE_ID.html#prop_tgt:UNITY_BUILD_UNIQUE_ID)).

Before my change the code didn't have `namespace {`, but it was supposed to have it even without UNITY_BUILD. So this PR also fixes a bunch of [ODR](https://en.wikipedia.org/wiki/One_Definition_Rule) violations.

cc @afuller-TT @ayerofieiev-tt

Reply:
Oops.  My bad.  I was thinking CMake would inject the sub-namespace for us.  But of course it won't.  It's just concatenating the files, not mucking with their contents.

But we are supposed to be having this tidbit:
```
// Needed for when unity builds are disabled
#ifndef MY_UNITY_ID
#define MY_UNITY_ID
#endif
```
when we leverage this feature.